pkg/cmd/pr/merge/merge.go

```diff
@@ -202,12 +202,13 @@ func mergeRun(opts *MergeOptions) error {
 	isPRAlreadyMerged := pr.State == "MERGED"
 	if !isPRAlreadyMerged {
 		payload := mergePayload{
-			repo:          baseRepo,
-			pullRequestID: pr.ID,
-			method:        opts.MergeMethod,
-			auto:          opts.AutoMergeEnable,
-			commitBody:    opts.Body,
-			setCommitBody: opts.BodySet,
+			repo:             baseRepo,
+			pullRequestID:    pr.ID,
+			mergeStateStatus: pr.MergeStateStatus,
```

Comment:
Instead of adding an additional field to `mergePayload`, could the caller figure out whether `auto` is necessary or not based on the value of `pr.MergeStateStatus`? I feel like this is more of a responsibility of the caller than the responsibility of the `mergePullRequest()` function.